airflow/providers/salesforce/operators/salesforce_bulk.py

```diff
@@ -0,0 +1,103 @@
+
+
+class SalesforceBulkOperator(BaseOperator):
+    """
+    Execute a Salesforce Bulk API and pushes results to xcom.
+
+    .. seealso::
+        For more information on how to use this operator, take a look at the guide:
+        :ref:`howto/operator:SalesforceBulkOperator`
+
+    :param operation: Bulk operation to be performed
+    :param object_name: The name of the Salesforce object
+    :param payload: list of dict to be passed as a batch
+    :param external_id_field: unique identifier field for upsert operations
+    :param batch_size: number of records to assign for each batch in the job
+    :param use_serial: Process batches in serial mode
+    :param salesforce_conn_id: The :ref:`Salesforce Connection id <howto/connection:SalesforceHook>`.
+    """
+
+    def __init__(
+        self,
+        *,
+        operation: str,
+        object_name: str,
+        payload: list,
+        external_id_field: str = 'Id',
+        batch_size: int = 10000,
+        use_serial: bool = False,
+        salesforce_conn_id: str = 'salesforce_default',
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.operation = operation
+        self.object_name = object_name
+        self.payload = payload
+        self.external_id_field = external_id_field
+        self.batch_size = batch_size
+        self.use_serial = use_serial
+        self.salesforce_conn_id = salesforce_conn_id
+
+    def execute(self, context: 'Context') -> dict:
+        """
+        Makes an HTTP request to Salesforce Bulk API.
+
+        :param context: The task context during execution.
+        :return: API response if do_xcom_push is True
+        :rtype: dict
+        """
+        available_operations = ['insert', 'update', 'upsert', 'delete']
+        if self.operation not in available_operations:
+            raise AirflowException(f'Operation not found! Available operations are {available_operations}')
+
+        sf_hook = SalesforceHook(salesforce_conn_id=self.salesforce_conn_id)
+        conn = sf_hook.get_conn()
+
+        if self.operation == 'insert':
+            result = conn.bulk.__getattr__(self.object_name).insert(
+                data=self.payload, batch_size=self.batch_size, use_serial=self.use_serial
+            )
+        elif self.operation == 'update':
+            result = conn.bulk.__getattr__(self.object_name).update(
+                data=self.payload, batch_size=self.batch_size, use_serial=self.use_serial
+            )
+        elif self.operation == 'upsert':
+            result = conn.bulk.__getattr__(self.object_name).upsert(
+                data=self.payload,
+                external_id_field=self.external_id_field,
+                batch_size=self.batch_size,
+                use_serial=self.use_serial,
+            )
+        else:
+            result = conn.bulk.__getattr__(self.object_name).delete(
+                data=self.payload, batch_size=self.batch_size, use_serial=self.use_serial
+            )
+
```

Comment:
Hey @ishiis, a simple suggestion. Add a validation on operators constructor about what operation will execute and don't execute the delete on `else`, instead of this, raise an exception.
If I what to update and make a mistake writing, `updat`, this operator will delete the passed ids.

Reply:
Check line 74

Reply:
Would be good to have a test case for this

Reply:
There is already a test case for this.

https://github.com/apache/airflow/pull/24473/files#diff-4db0aba80d593c6846d4ab4b91327f59986e7b7e304aa0cc0e6df9b85f76b0ebR32-R44

Reply:
@mariotaddeucci 
I moved the validation to else clause. This is better.

Reply:
Both are good, a validation on constructor will help to find errors when the dag is generated, not on execution time, becoming easier to get errors on unit tests for example

Reply:
@mariotaddeucci 
I moved the validation to constructor and add validation for object_name.